### ISSUE TITLE
Group patterns "" and "/*"  with "/contents" for the contents API 

### ISSUE THREAD

**Issue by bharatnc** [gogs/gogs#5985]
**Describe the bug**
 TLDR:  `/contents/*` and `/contents` should both work (Github API v3 works this way)

The contents api should default to the default repository path if `:path` is not present. 
Example:
The following should work and point to the contents at the root of the repo
```bash
http://try.gogs.io/api/v1/repos/testuser/testrepo/contents?ref=master
```

The following should also work and should point to the actual path specified if exists and is valid.
```bash
http://try.gogs.io/api/v1/repos/testuser/testrepo/contents/foo/bar?ref=master
```
Github API v3 works in the above described way: 

```bash
 gogs git:(master) ✗ curl -H "Authorization: token REDACTED" https://api.github.com/repos/bharatnc/test-repo/contents | jq . 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  4091  100  4091    0     0   9283      0 --:--:-- --:--:-- --:--:--  9276
[
  {
    "name": "dir1",
    "path": "dir1",
    "sha": "b4f457c219dbb3517be908d4e70f0ada2fd8b8f9",
    "size": 0,
    "url": "https://api.github.com/repos/bharatnc/test-repo/contents/dir1?ref=master",
    "html_url": "https://github.com/bharatnc/test-repo/tree/master/dir1",
 
..... 
OUTPUT TRUNCATED
```

**Gogs version or commit**

On master.

**Expected behavior**

Without providing the `:path` it should automatically point to the root of the repo.

**Actual behavior**
Currently just throws 404 path or not found error.

**Fix**

The fix is actually quite simple in this case. Just add group `"contents"` with `""` and `"/*"`, in `gogs.io/gogs/internal/route/api/v1/api.go`.

```
diff --git a/internal/route/api/v1/api.go b/internal/route/api/v1/api.go
index b54a6df..9839404 100644
--- a/internal/route/api/v1/api.go
+++ b/internal/route/api/v1/api.go
@@ -271,7 +271,10 @@ func RegisterRoutes(m *macaron.Macaron) {
                                }, reqRepoAdmin())
 
                                m.Get("/raw/*", context.RepoRef(), repo.GetRawFile)
-                               m.Get("/contents/*", repo.GetContents)
+                               m.Group("/contents", func() {
+                                       m.Get("", repo.GetContents)
+                                       m.Get("/*", repo.GetContents)
+                               })
                                m.Get("/archive/*", repo.GetArchive)
                                m.Group("/git/trees", func() {
                                        m.Get("/:sha", repo.GetRepoGitTree)
(END)

```

**Comment by bharatnc** [comment]
I have actually tested the fix locally and I will open a PR shortly.

**Comment by bharatnc** [comment]
> I have actually tested the fix locally and I will open a PR shortly.

@unknwon  https://github.com/gogs/gogs/pull/5986 thanks!
